### PR TITLE
HW Spring Boot Data JPA

### PR DESCRIPTION
**Pull request by savchenkoarts**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -32,4 +38,17 @@ public BookDto getBookById(@PathVariable Long id) {
...
     }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    public void isDeleted(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<BookDto> updateBook(@PathVariable Long id,
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/onlinebookstore/controller/BookController.java:49]: redundant wrapping 
```suggestion
    public BookDto updateBook(@PathVariable Long id,
```

File: src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -13,4 +13,5 @@ public class CreateBookRequestDto {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    private boolean isDeleted;
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java:16]: Don’t  pass it like a request parameter

File: src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -37,4 +39,22 @@ public BookDto createBook(CreateBookRequestDto createBookRequestDto) {
...
+    public BookDto updateBook(Long id, UpdateBookRequestDto updateRequest) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Book with id " + id + " not found"));
+        Optional.ofNullable(updateRequest.getTitle()).ifPresent(book::setTitle);
+        Optional.ofNullable(updateRequest.getAuthor()).ifPresent(book::setAuthor);
+        Optional.ofNullable(updateRequest.getIsbn()).ifPresent(book::setIsbn);
+        Optional.ofNullable(updateRequest.getPrice()).ifPresent(book::setPrice);
+        Optional.ofNullable(updateRequest.getDescription()).ifPresent(book::setDescription);
+        Optional.ofNullable(updateRequest.getCoverImage()).ifPresent(book::setCoverImage);
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java:57]: use the method of mapper 
`void updateBook(Book book, UpdateBookRequestDto updateRequest)`

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-
-spring.jpa.show-sql=true
-spring.jpa.hibernate.ddl-auto=create-drop
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.open-in-view=false
-
+spring.jpa.show-sql=true
 server.servlet.context-path=/api
+spring.liquibase.change-log=classpath:db/changelog/db.changelog-master.yaml
```
Comment (liliia-ponomarenko) [inline src/main/resources/application.properties:9]: do you really need it? Looks like a default path

File: src/main/resources/db/changelog/changes/01-create-book-table.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                  name: price
+                  type: DECIMAL(19,2)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: description
+                  type: TEXT
+              - column:
+                  name: cover_image
+                  type: VARCHAR(255)
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/01-create-book-table.yaml:40]: add one empty line (check other files)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s improve your solution ;)

**Review by Sanderfuck — APPROVED** [review]

**Pull request merged**